src/mon/OSDMonitor.cc

```diff
@@ -10756,6 +10824,16 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	    ss << "marked down osd." << osd << ". ";
 	    any = true;
 	  }
+	  if (definitely_dead) {
+	    if (!pending_inc.new_xinfo.count(osd)) {
+	      pending_inc.new_xinfo[osd] = osdmap.osd_xinfo[osd];
+	    }
+	    if (pending_inc.new_xinfo[osd].dead_epoch < pending_inc.epoch) {
+	      any = true;
+	    }
+	    pending_inc.new_xinfo[osd].dead_epoch = pending_inc.epoch;
+	    derr << "setting dead_epoch" << dendl;
```

Comment:
If we want to leave this in, it probably should include osd and pending_inc.epoch.